src/Core/Util/Url/UrlCleaner.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Util\Url;
+
+class UrlCleaner
```

Comment:
Because it's an internal tool used by the Core, there are already other similar tools Where else do you think it'd better be? It shouldn't be in Adapter as it's not temporary nor adapting legacy to core. It's not really linked to Symfony framework so it shouldn't be in PrestaShopBundle either so Core seemed like the right place

I'm open to alternatives though? But which one?

Reply:
My opinion is that creating a bundle doesn't mean you should only have classes linked to Symfony Framework but the features provided should be linked to Symfony features. UrlCleaner it's just a helper not an api used elsewhere since the only usage in the PrestaShopBundle folder itself (in TokenizedUrlsListener class). Said another way, it's an Internal class of the PrestaShopBundle. We could have just used a private method inside TokenizedUrlsListener class and the question would not happened. 
That said, I think this class should live in the PrestaShopBundle folder (for exemple in Utils) IMO.

Reply:
It can also be used outside of the core, inside modules. To me, it fits perfectly in the Core namespace.

Reply:
I don't think creating a new file is needed, because it does not seems to be used or useful somewhere else, this should be a private method IMO as this is only internal feature.

First I was confused about it being static and not injected, but in the end ~it does not even matter~ I think we should think simple and not overthink things (YAGNI ;) ), this should be a private method IMO as it is only used once and is for internal use. Also, that avoids discussions about where to place it 😅 and having to deal with BC promise.